src/subapps/admin/components/Settings/DangerZoneSubView.tsx

```diff
@@ -130,6 +161,39 @@ const DangerZoneSubView = ({ project }: Props) => {
       },
     }
   );
+
+  const undoDeprecateProjectAsync = async (
+    apiEndpoint: string,
+    nexus: NexusClient,
```

Comment:
I suggest to use `useMutation` from react-query 
1. get the loading state pass it to the action button
2. can handle success/failure part more clean
3. if we are fetching the project details with RQ, then you can do the refresh just by invalidating the cache

Reply:
Done.